pkg/operator/controllers/etchosts/machineconfig_controller.go

```diff
@@ -141,7 +156,7 @@ func (r *EtcHostsMachineConfigReconciler) SetupWithManager(mgr ctrl.Manager) err
 		Complete(r)
 }
 
-func reconcileMachineConfigs(ctx context.Context, instance *arov1alpha1.Cluster, role string, dh dynamichelper.Interface, mcps ...mcv1.MachineConfigPool) error {
+func reconcileMachineConfigs(ctx context.Context, instance *arov1alpha1.Cluster, role string, dh dynamichelper.Interface, c client.Client, allowReconcile bool, mcps ...mcv1.MachineConfigPool) error {
```

Comment:
> allowReconcile only matters on calls to Ensure

It's true,

> I think if we wanted to make this more clear, we could have some follow-up work to move Ensure into Reconcile (probably for every controller - not just this one), with all other functions building up a resources object to pass into Ensure - how does that sound?

 but I don't understand why we need to move `Ensure` to `Reconcile`.
I just think that the logic from `r.AllowRebootCausingReconciliation(ctx, instance)` to `Ensure` is not used if `allowReconcile` is false (sorry if this understanding is wrong).
If it stops reconciling just after `r.AllowRebootCausingReconciliation(ctx, instance)`, it will be more readable because we don't have to follow the lines until `Ensure`, and also there will be (very) slight improvement in performance.

Also `AllowRebootCausingReconciliation` is used only in dnsmasq and etchosts.
And the dnsmasq has the reason not to stop earlier (because it has to create some things), so I don't think we have the convention we should follow here.

> We can card this up but to my knowledge this allowReconcile approach is going to be EOL with SyncSets. Eventually this will move into syncsets.

If so, we might not necessarily have to change the code, but I'd say just changing the code here is doable and worth doing.

Reply:
> You seem to had a fundamental mistunderstanding of what allowReconcile is doing. This must be used to stop from Ensure or Create being called to have the desired effect.

That's what I understand.
That's why I'm asking you why these codes are needed to run even though they don't Ensure or Create.

https://github.com/Azure/ARO-RP/blob/3762a791fe70e5ac439d03d289b31a3dacdddf86/pkg/operator/controllers/etchosts/machineconfig_controller.go#L160-L178